File: app/src/main/java/com/makingiants/today/todayhistory/utils/Transformer.java
package com.makingiants.today.todayhistory.utils;

import rx.Observable;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Transforms to avoid setting each Schedulers on each call.
 * Use:
 *
 * observable.compose(applyIoSchedulers())
 */
public class Transformer {
  public static <T> Observable.Transformer<T, T> applyIoSchedulers() {
    return observable -> observable.subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread());
  }
}

File: app/src/prod/kotlin/com/makingiants/todayhistory/utils/Transformer.kt
package com.makingiants.todayhistory.utils

import rx.Observable
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

/**
 * Transforms to avoid setting each Schedulers on each call.
 * Use:

 * observable.compose(applyIoSchedulers())
 */
object Transformer {
    fun <T> applyIoSchedulers(): Observable.Transformer<T, T> = Observable.Transformer<T, T> {
        it.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
    }
}